### ISSUE TITLE
Improving the WaveDiff optimization v2: New scenario

### ISSUE THREAD

**Issue by tobias-liaudat** [tobias-liaudat/wf-SEDs#19]
In this scenario, the parametric model can reproduce the GT model. In other words, the GT model is contained in the space of models spanned by our parametric model.

We would like to see if the non-parametric part can help the optimisation of the parametric part.

## Dataset

Same `Dataset n1` as in #17 

Parameters: WFE_res(GT)=256, n_bins(GT)=350, n_stars=2000.
Inputs: Possibility to use n_bins=8, 16, 32. We will use n_bins=8 and WFE_res=256 for the models.

## Models

- [x] `Model n11 bis`:  **without projecting** `model =poly`, `d_max_nonparam=5`, `n_zernikes =45` and `d_max=2`. Use `2 cycles` without projecting the last cycle. 
- [x] `Model n11.2`: **without projecting** `model =poly`, `d_max_nonparam=5`, `n_zernikes =60` and `d_max=2`. Use `2 cycles` without projecting the last cycle.

- [x] `Model n12`: **projecting** `model =poly`, `d_max_nonparam=5`, `n_zernikes =45` and `d_max=2`. Use `5 cycles` without projecting the last cycle. (25)
- [x] `Model n12 bis`:  **without projecting** `model =poly`, `d_max_nonparam=5`, `n_zernikes =45` and `d_max=2`. Use `5 cycles` without projecting the last cycle. (25)

- [x] `Model n12.1`: **projecting** `model =poly`, `d_max_nonparam=5`, `n_zernikes =45` and `d_max=5`. Use `5 cycles` without projecting the last cycle. (25)
- [x] `Model n12.2`: **projecting** `model =poly`, `d_max_nonparam=5`, `n_zernikes =60` and `d_max=2`. Use `5 cycles` without projecting the last cycle. (25)

- [x] `Model n13`: **projecting** `model =poly`, `d_max_nonparam=5`, `n_zernikes =45` and `d_max=2`. Use `9 cycles` without projecting the last cycle. (25)
- [x] `Model n13 bis`:  **without projecting** `model =poly`, `d_max_nonparam=5`, `n_zernikes =45` and `d_max=2`. Use `9 cycles` without projecting the last cycle. (25)

## Statistical analysis
- [x] Repeat optimisation for every model above by performing 5 realisations.
- [x] Evaluate the models with and without the non-parametric part. Project the last cycle NP features if `project_dd_features = True`.
- [x] Evaluate the models with and without the non-parametric part. Project the last cycle NP features for every model.

## New experience

In this experience, we only train the non-parametric part. After a cycle, the non-parametric part projects over the parametric part, and for some models it is initialised following the standard initialization procedure.

Model parameters: `model =poly`, `d_max_nonparam=5`, `n_zernikes =45` and `d_max=2`. Number of non-parametric epochs per cycle: `75`, non-param learning rate: `0.1`. Train **only the non parametric model**

### Scenarios to test:

- [x] `n_cycles = 2`, `reset_dd_features = False`, `eval_only_param = True`.
- [x] `n_cycles = 2`, `reset_dd_features = False`, `eval_only_param = False`.
- [x] `n_cycles = 2`, `reset_dd_features = True`, `eval_only_param = True`.
- [x] `n_cycles = 2`, `reset_dd_features = True`, `eval_only_param = False`.
- [x] `n_cycles = 5`, `reset_dd_features = False`, `eval_only_param = True`.
- [x] `n_cycles = 5`, `reset_dd_features = False`, `eval_only_param = False`.
- [x] `n_cycles = 5`, `reset_dd_features = True`, `eval_only_param = True`.
- [x] `n_cycles = 5`, `reset_dd_features = True`, `eval_only_param = False`.

- [x] Present results in a table
- [x] Analyse the optimisation curves

### Additional scenarios:
- [x] `n_cycles = 9`, `reset_dd_features = True`, `eval_only_param = True`.
- [x] `n_cycles = 9`, `reset_dd_features = True`, `eval_only_param = False`.

- [x] `n_cycles = 5`, `reset_dd_features = True`, `eval_only_param = True`. Using only **15 epochs** per cycle.
- [x] `n_cycles = 5`, `reset_dd_features = True`, `eval_only_param = False`. Using only **15 epochs** per cycle.
- [x] `n_cycles = 9`, `reset_dd_features = True`, `eval_only_param = True`. Using only **15 epochs** per cycle.
- [x] `n_cycles = 9`, `reset_dd_features = True`, `eval_only_param = False`. Using only **15 epochs** per cycle.

### Testing the pretrained parametric model.

- [ ] Take the best performing model (WFE-wise) and perform 2 parametric training cycles. 
- [ ] **Sanity check**: train only the parametric model from scratch in the same way it's trained in the paper.

## Back to alternate optimisation

Optimize the model alternating the parametric and the non-parametric models as before but now the non-parametric model is reseted after projecting at the end of every cycle. Using: `n_epochs_param = 15` and `n_epochs_nonparam = 75` with `llr_param = 0.004` and `llr_nonparam=0.1`.

- [x] `n_cycles = 5`, `reset_dd_features = True`, `eval_only_param = True`.
- [x] `n_cycles = 9`, `reset_dd_features = True`, `eval_only_param = False`.

**Comment by CentofantiEze** [comment]
All the new scenarios and the parametric-pretrained model are queued on JZ.

**Comment by tobias-liaudat** [comment]
Done
